caikit/runtime/http_server/http_server.py

```diff
@@ -679,6 +753,126 @@ async def _handler(
+
+    def _add_prediction_job_unary_input_handler(self, rpc: TaskPredictionJobRPC):
+        """Add a unary:unary request handler to start a prediction job for this RPC"""
+
+        # Get request variables in same way as unary_unary
+        pydantic_predict_request = dataobject_to_pydantic(
+            self._get_request_dataobject(rpc)
+        )
+        predict_request_openapi = self._get_request_openapi(pydantic_predict_request)
+
+        # Result will always be a Prediction Job
+        pydantic_job_response = dataobject_to_pydantic(PredictionJob)
+        job_response_openapi = self._get_response_openapi(
+            PredictionJob, pydantic_job_response, 202
+        )
+
+        # Merge the DataObject openapi schema into the task schema
+        task_api_schema = merge_configs(
+            rpc.task.get_metadata().get(EXTRA_OPENAPI_KEY, {}), predict_request_openapi
+        )
+
+        @self.app.post(
+            get_http_prediction_job_route_name(rpc.name),
+            responses=job_response_openapi,
+            include_in_schema=rpc.task.get_visibility(),
+            description=rpc.task.__doc__,
+            openapi_extra=task_api_schema,
+            response_class=Response,
+        )
+        # pylint: disable=unused-argument
+        async def _handler(
+            context: Request,
+        ) -> Response:
+            try:
+                request = await pydantic_from_request(pydantic_predict_request, context)
+                request_params = self._get_request_params(rpc, request)
+
+                model_id = self._get_model_id(request)
+                log.debug4(
+                    "Starting prediction job with args %s for model id %s",
+                    request_params,
+                    model_id,
+                )
+
+                # After fetching the model_id from the request, notify module
+                # backends of the request context which may influence the lazy
+                # initialization logic.
+                self.global_predict_servicer.notify_backends_with_context(
+                    model_id, context
+                )
+                log.debug(
+                    "In prediction job handler for %s for model %s", rpc.name, model_id
+                )
+                loop = asyncio.get_running_loop()
+
+                call = partial(
+                    self.global_predict_servicer.run_prediction_job,
+                    model_id=model_id,
+                    request_name=rpc.request.name,
+                    task=rpc.task,
+                    context=context,
+                    **request_params,
+                )
+                result = await loop.run_in_executor(self.thread_pool, call)
+                log.debug4("Job started from model %s with id", model_id, result.job_id)
+
+                return Response(
+                    content=result.to_json(),
+                    media_type="application/json",
+                    status_code=status.HTTP_202_ACCEPTED,
```

Comment:
Same point here about 202. This isn't an API breaking change since it's a new endpoint, but we would want consistency.